src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-param-reassign */
+import { Post } from '../../types/Post';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getUserPosts } from '../../api/posts';
+import { RootState } from '../../app/store';
+
+export const loadPosts = createAsyncThunk(
+  'posts/fetchPosts',
+  (userID: number) => {
+    return getUserPosts(userID);
+  },
+);
+
+const initialState = {
+  posts: [] as Post[],
+  loaded: false,
+  hasError: false,
+};
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setPosts: state => {
+      state.posts = [];
```

Comment:
The `setPosts` reducer resets the posts to an empty array. Verify if this is the intended behavior. If you need to set posts with specific data, consider modifying the reducer to accept a payload and update the state accordingly.